src/js/game/LevelMVC/LevelView.js

```diff
@@ -1979,4 +1997,22 @@ module.exports = class LevelView {
     this.resettableTweens.push(tween);
     return tween;
   }
+
+  /**
+  * Animate Door and set the status
+  */
+  animateDoor(index, Open) {
+    this.controller.levelModel.actionPlane[index].isOpen = Open;
+    let player = this.controller.levelModel.player;
+    this.setSelectionIndicatorPosition(this.controller.levelModel.actionPlane.indexToCoordinates(index)[0], this.controller.levelModel.actionPlane.indexToCoordinates(index)[1]);
+    this.controller.audioPlayer.play("doorOpen");
+    // If it's not walable, then open otherwise, close.
+    const canOpen = !this.controller.levelModel.actionPlane[index].isWalkable;
```

Comment:
I'm a bit confused by this. We're setting the door's open state based on the argument being passed here, but then deciding which animation to play based on its walkability? Seems like we're putting ourselves at risk of these two concepts getting out of sync. What am I missing here?

Reply:
Much like the other animation logic above, this was taken wholesale from the GameController and the way that door handling was done there. isOpen was added for door specific logic with pressure-plate stuff. I would wager that we can replace the walkable check with the open check. I didn't alter too much of that so much have just missed the check. 👍